Disallow robots from indexing documentation pages

@virtualtam [wrote](https://github.com/shaarli/Shaarli/issues/930#issue-252205772):


>Also, there should be a robots.txt under `doc/html` to prevent bots from indexing users's docs: [Google search on "shaarli server requirements"](https://encrypted.google.com/search?hl=en&q=server+requirements+shaarli&oq=server+requirements+shaarli&gs_l=psy-ab.3...25733.28299.0.28711.20.12.0.0.0.0.377.1808.0j1j5j1.7.0....0...1.1.64.psy-ab..18.0.0.BBvN2PvSA-w)

> On second thoughts, I'm not sure how it would behave when running Shaarli behind one or more reverse proxies?

> Maybe it should be handled through the HTTP server / proxy configuration as shown in https://stackoverflow.com/questions/1090823/how-do-i-configure-nginx-to-redirect-to-a-url-for-robots-txt-sitemap-xml

@nodiscc [wrote](https://github.com/shaarli/Shaarli/issues/930#issuecomment-403235249):

> See https://stackoverflow.com/questions/38305272/robots-txt-for-application#38305597

>Since we cannot know the desired robots.txt configuration for every instance (administrator may want to set custom robots rules, shaarli may be installed alongside other applications, etc), I think it is best not touch the server's configuration regarding this file. What we could do is:

> - Document robots.txt usage on the Server Configuration page in the docs
> - Disallow indexing of doc pages via `<meta>`-tags as suggested in the SO answer above


https://developers.google.com/search/reference/robots_meta_tag

Following https://github.com/shaarli/Shaarli/pull/1220, building the docs fails with https://readthedocs.org/projects/shaarli/builds/7886340/

```
  File "/home/docs/checkouts/readthedocs.org/user_builds/shaarli/checkouts/master/doc/custom_theme/main.html", line 23, in template
    {% if config.site_favicon %}<link rel="shortcut icon" href="{{ config.site_favicon|url }}">
[...]
jinja2.exceptions.TemplateAssertionError: no filter named 'url'
```

I will have time to fix it today

--- a/doc/custom_theme/main.html
+++ b/doc/custom_theme/main.html
@@ -20,6 +20,4 @@ the blocks defined in base.html and its included child templates.
 
 {% if page and page.is_homepage %}<meta name="description" content="{{ config.site_description }}">{% endif %}
 {% if config.site_author %}<meta name="author" content="{{ config.site_author }}">{% endif %}
-{% if config.site_favicon %}<link rel="shortcut icon" href="{{ config.site_favicon|url }}">
-{% else %}<link rel="shortcut icon" href="{{ 'img/favicon.ico'|url }}">{% endif %}
 {%- endblock %}

